### PR TITLE
Create Themes Modal

### PR DESCRIPTION
**Pull request by jcheringer**
## Summary

The purpose of this PR is to allow the user to change their current project theme.
On the Document tab from the right sidebar, there will be a `Theme` section that will allow the user to open the theme selection modal.

Depends on: D76409-code

Ref: c/iU4SPLeB-tr

## Test Instructions
* Open a project and, on the right sidebar, under the `Theme` section, click the `Change Theme` button
* You should see the theme selection modal
* The modal should show your current active theme and allow you to change the project theme to one of the available themes

## Screenshots

image

image

### CODE REVIEW

**Comment by jcheringer** [comment]
Known issues:
* Pages previews, on the left side, are not updating after changing the theme;
* By changing the theme, the `Publish` button should become available to update the Public version theme;

**Review by digitalwaveride — CHANGES_REQUESTED** [review]
Very very cool João! 😍 
So fun to switch between the themes! 🥳 

### Creation of the project behavior

When creating a new project, the "Theme" section in the Document settings tab is not yet available... neither with an empty project nor with a template as a start.

image

image

Only after saving the draft does it show.
Anything we can do about it?

Wondering how many users would do a "change theme" action very early in the project... or might look for themes area in the very beginning...  


### Modal Window
Two little things I noticed in the modal window... strangely the behavior is always related to which theme is activated:

#### A)

When the TwentyNineteen theme is activated, and only then, the font below the theme previews are not loading correctly 🤷 

image

#### B)

When any other theme besides the Leven theme is activated, the button "Activate this theme" is missing on hover the "hand" the "pointer cursor state".
(pointer appears on these buttons only when Leven theme is activated 🤷 )


### Quadrat Theme

The quadrat is unfortunately not rendering for me ... 

image


### Preview Column... 

As you mentioned, the preview column is not adjusting to new theme ... known issue

image



Otherwise, looking good!!!

**Comment by jcheringer** [comment]
Hey @digitalwaveride, thanks for testing it!
From the issues you reported, I believe I handled all of them, except for the Quadrat theme and the preview column.
The Quadrat PR is still under revision on #178.
And, the preview still needs investigation.
If everything else looks good, I'd suggest handling the preview issue on another PR since this one is already pretty big. 😬

**Comment by jcheringer** [comment]
The [Publish button issue](https://github.com/Automattic/crowdsignal-ui/pull/179#issuecomment-1065130284) was also fixed.

**Comment by CGastrell** [comment]
Not sure what's happening, but as soon as the the project attempts a save the editor crashes with this error at the bottom:
image

Apparently coming from `<DocumentSettings>` component. Doesn't this happen to you? I applied the backend patch as well

File: apps/project-renderer/src/components/app/index.js [hunk 1]
```diff
@@ -27,9 +27,9 @@ const App = ( {
 	preview,
 	respondentId = '',
 	startTime = 0,
-	theme = 'leven',
 } ) => {
 	const [ content, setContent ] = useState( [] );
+	const [ theme, setTheme ] = useState( null );
```
[the remark this answers is not in the document]
Comment (jcheringer) [reply apps/project-renderer/src/components/app/index.js:32]: Well, I think this is something we could explore further.

[the remark this answers is not in the document]
Comment (ice9js) [reply apps/project-renderer/src/components/app/index.js:32]: +1. It does make sense for the project renderer where the themes don't change dynamically.  
That said, we still need something for local testing so probably the best solution is to wrap it up somehow and put it behind one of those `process.env.NODE_ENV === 'production'` flags.

**Pull request merged**